### ISSUE TITLE
from __future__ import annotations doesn't ignore annotations correctly

### ISSUE THREAD

**Issue by Starwort** [PyCQA/pylint#3236]
### Steps to reproduce
1. Use code
```py
from __future__ import annotations
from collections import defaultdict
squares: defaultdict[tuple, int] = defaultdict(lambda: 999999)
```
2. Run pylint on this code
3. `E1136: Value 'defaultdict' is unsubscriptable (unsubscriptable-object)`

### Current behaviour
Throws `E1136: Value 'defaultdict' is unsubscriptable (unsubscriptable-object)`

### Expected behaviour
Ignores it (as Python itself does)

### pylint --version output
```
pylint 2.4.3
astroid 2.3.1
Python 3.8.0 (default, Oct 15 2019, 14:05:58) 
[GCC 7.4.0]
```

**Comment by PCManticore** [comment]
Thanks for opening an issue! This error makes sense somewhat as you should use `typing.DefaultDict` instead. Even `mypy` emits the same warning with:
```
a.py:3: error: "defaultdict" is not subscriptable, use "typing.DefaultDict" instead
```

While Python ignores the variable annotations, they can still contain errors such as this one.

**Comment by Starwort** [comment]
I submitted a near-identical issue to mypy, as the tools should still accept these as valid, even if they aren't preferred - as they are valid typehints. Disagree with closure

**Comment by Starwort** [comment]
Furthermore, here is the issue, complete with MyPy's feature and priority tags, that I submitted to MyPy (along with confirmation this is being added; i.e. MyPy will stop reporting this error).
https://github.com/python/mypy/issues/7907#issuecomment-551896196

**Comment by Starwort** [comment]
https://www.python.org/dev/peps/pep-0585/
PEP 585 explicitly allows this behaviour. I'd like this issue reopened accordingly; I may open a new issue if I think this isn't noticed :P
